### PR TITLE
chore(deps): upgrade fvm deps

### PR DESCRIPTION
**Pull request by hanabi1224**
## Summary of changes

As part of https://github.com/ChainSafe/forest/issues/3422

> We might want to start integrating Forest and fil-actor states with the new `ref-fvm`

Changes introduced in this pull request:

- Upgrade `fvm*` crates
- Upgrade `fil_actor_*` crates
- Make `fvm_ipld_bitfield`, `fvm_ipld_amt` and `fvm_ipld_hamt` indirect dependencies by using re-exports from `fil_actors_shared` to make future upgrades easier

Manually tested on DO droplet, forest has been running on mainnet over the weekend with no problem.

## Reference issue to close (if applicable)

Closes https://github.com/ChainSafe/forest/issues/3478

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: src/tool/subcommands/snapshot_cmd.rs [hunk 1]
```diff
@@ -618,6 +618,8 @@ mod structured {
                         method_num,
                         params,
                         value,
+                        gas_limit: _,
```
Comment (hanabi1224) [inline src/tool/subcommands/snapshot_cmd.rs:621]: Question: does lotus json render these 2 field? @aatifsyed

File: src/state_manager/chain_rand.rs [hunk 2]
```diff
@@ -221,11 +189,17 @@ pub fn draw_randomness(
...
     state.write_i64::<BigEndian>(round)?;
     state.write_all(entropy)?;
     let mut ret = [0u8; 32];
     ret.clone_from_slice(state.finalize().as_bytes());
     Ok(ret)
 }
+
+/// Computes a 256-bit digest.
+/// See <https://github.com/filecoin-project/ref-fvm/blob/master/fvm/CHANGELOG.md#360-2023-08-18>
+pub fn digest(rbase: &[u8]) -> [u8; 32] {
```
Comment (LesnyRumcajs) [inline src/state_manager/chain_rand.rs:203]: what is `rbase`?

**Review by LesnyRumcajs — APPROVED** [review]
Fantastic job.

File: src/state_manager/chain_rand.rs (hunk 2 above)
Comment (hanabi1224) [reply src/state_manager/chain_rand.rs:203]: I guess it mean `base` array for `r`andomness

**Pull request merged**